wp-content/plugins/star-wars/index.php

```diff
@@ -1,3 +1,130 @@
 <?php
 
-//STAR WARS PLUGIN GOES HERE
\ No newline at end of file
+/**
+ * @package Start-Wars
+ * @version 1.0
+ */
+/*
+Plugin Name: Star Wars
+Description: This plug-in is a challenge to apply for Developer Job in StickerStock. The plugin allows to randomly show in the any page top a Star Wars character from a list getting from Star Wars API https://swapi.dev/. 
+Author: Marco Portero
+Version: 1.0
+Author URI: https://www.linkedin.com/in/marcoportero/
+*/
+
+function star_wars_get_people() {
+
+	$people = 'Nobody';
+    // Get all characters from Star Wars API
+	$url = 'https://swapi.dev/api/people';
+
+      $curl = curl_init();
+      curl_setopt_array($curl, array(
+        CURLOPT_URL => $url,
+        CURLOPT_RETURNTRANSFER => true,
+        CURLOPT_FOLLOWLOCATION => true,
+        CURLOPT_ENCODING => "",
+        CURLOPT_MAXREDIRS => 10,
+        CURLOPT_SSL_VERIFYPEER => false,
+        CURLOPT_SSL_VERIFYHOST => false,  
+        CURLOPT_TIMEOUT => 30,
+        CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+        CURLOPT_CUSTOMREQUEST => "GET"
+      ));
+      $response = curl_exec($curl);
+      $err = curl_error($curl);
+      curl_close($curl);
+      if ($err) {
+        //Only show errors while testing
+        printf( "cURL Error #: %s", $err);
+
+      } else {
+        
+        $responseObj = json_decode($response);
+        $people = $responseObj->results;
+        
+      }
+    
+    // Get a random position from the array
+    $pos = mt_rand( 0, count( $people ) - 1 );
+
+	// And then randomly choose a star wars character
+	return wptexturize( $people[ $pos ]->name . " | Gender: " . $people[ $pos ]->gender);
+	//return wptexturize( $people );
+}
+
+// Prepare data for show it.
+function star_wars() {
+	$chosen = star_wars_get_people();
+	
+	$lang   = '';
+	if ( 'en_' !== substr( get_user_locale(), 0, 3 ) ) {
+		$lang = ' lang="en"';
+	}
+
+	printf(
```

Comment:
Nicely done! 

But keep in mind that StickerStoke focuses on 2D and 3D graphics primarily. 

It's not like a banner is bad, but a text banner is definitely a ⚠️

Hint: the `planets/` endpoint provides a wider range of information that could be visually displayed in a simple way (eg: colour codes or SVG icons)

Reply:
The widget is a nice touch! But I think a practical example will be better to explain what's expected:

By doing `$chosen = star_wars_get_people();` we can get `$chosen->eye_color` and this will give me a string with the color name.

By using this [Gist](https://gist.github.com/kiley0/756bf555c5a7ae17b8d03596ae364712#gistcomment-3780302) we can temporarily hardcode an object map to get a valid colour code to visually modify anything (eg: background colour, border colour, SVG path with eye shape that would only change the eyes' iris based on that colour map 💪🏼 ).

It's crucial to display textual information. But we need to understand the capabilities to eventually develop things that are more [visual/interactive](https://blackstars.surge.sh/).